src/ukify/ukify.py

```diff
@@ -997,6 +1028,45 @@ def pe_add_sections(uki: UKI, output: str) -> None:
             pe.__structures__.append(new_section)
             pe.sections.append(new_section)
 
+    # If there is a pre-signed JSON blob, we need to update the existing JSON, by appending the signature to
+    # each corresponding digest object. We have built the unsigned UKI with enough space to fit the .sig
+    # objects, so we can just replace the new signed JSON in the existing sections.
+    if opts.pcrsig:
+        signatures = json.loads(str(opts.pcrsig))
+        for i, section in enumerate(pe.sections):
+            if pe_strip_section_name(section.Name) == '.pcrsig':
+                j = json.loads(
+                    bytes(
+                        pe.__data__[
+                            section.PointerToRawData : section.PointerToRawData + section.SizeOfRawData
+                        ]
+                    )
+                    .rstrip(b'\x00')
+                    .decode()
+                )
+                for bank, sigs in j.items():
+                    for sig in sigs:
+                        for input_bank, input_sigs in signatures.items():
+                            for input_sig in input_sigs:
+                                if input_bank == bank and sig['pol'] == input_sig['pol']:
+                                    sig['sig'] = input_sig['sig']
```

Comment:
That's a lot of indentation. This should skip a couple of iterations
```suggestion
                for bank, sigs in j.items():
                    for input_bank, input_sigs in signatures.items():
                        if bank != input_bank:
                            continue
                        for sig in sigs:
                            for input_sig in input_sigs:
                                if sig['pol'] == input_sig['pol']:
                                    sig['sig'] = input_sig['sig']
```
or maybe even
```suggestion
                for (bank, sigs), (input_bank, input_sigs) in itertools.product(
                    j.items(), signatures.items()
                ):
                    if input_bank != bank:
                        continue
                    for sig, input_sigs in itertools.product(sigs, input_sigs):
                        if sig['pol'] == input_sig['pol']:
                            sig['sig'] = input_sig['sig']
```